bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/UpnpDiscoveryService.java

```diff
@@ -102,6 +105,11 @@ protected void startScan() {
         }
         upnpService.getRegistry().addListener(this);
         upnpService.getControlPoint().search();
+
+        // Belkin WeMo devices can not be discovered by default search() function, but have to be searched by their
+        // named service.
+        ServiceType wemoBasicEvent = new ServiceType("Belkin", "basicevent");
+        upnpService.getControlPoint().search(new ServiceTypeHeader(wemoBasicEvent));
```

Comment:
I do not like that this introduces WeMo specific code in the general upnp discovery. This will be executed, even if the WeMo binding is not installed which should not be the case.
One way around that would be to add a new method to the UPnPDiscoveryParticipant (like getServiceTypeHeadersForBroadcast()), which could then be iterated over in this line.

Does anyone has any other idea how to decouple it best?